### PR TITLE
Fix notifier relabel changing the labels of active alerts

### PR DESCRIPTION
**Pull request by codesome**
Closes https://github.com/prometheus/prometheus/issues/11424

### CODE REVIEW

**Comment by codesome** [comment]
@roastiek would you be able to try this fix?

**Comment by codesome** [comment]
@roastiek nevermind, I was able to reproduce it.

File: rules/alerting.go [hunk 1]
```diff
@@ -507,6 +507,8 @@ func (r *AlertingRule) sendAlerts(ctx context.Context, ts time.Time, resendDelay
 			}
 			alert.ValidUntil = ts.Add(4 * delta)
 			anew := *alert
+			// The notifier re-uses the labels slice, hence make a copy.
+			anew.Labels = alert.Labels.Copy()
```
Comment (codesome) [inline rules/alerting.go:511]: This is the actual fix. Rest is some refactor to write the unit test.

**Review by roidelapluie — APPROVED** [review]
LGTM

File: rules/manager.go [hunk 2]
```diff
@@ -35,9 +35,11 @@ import (
 	"github.com/prometheus/prometheus/model/rulefmt"
 	"github.com/prometheus/prometheus/model/timestamp"
 	"github.com/prometheus/prometheus/model/value"
+	"github.com/prometheus/prometheus/notifier"
```
Comment (bboreham) [inline rules/manager.go:38]: I wonder if having `sendAlerts` in the main package was intended to separate `manager` from depending on `notifier` ?

Comment (codesome) [reply rules/manager.go:38]: Possible. We can revisit this again. It can be moved to a util package.

**Pull request merged**